Update documentation for Subscription v1alpha2 CRD (#16340)
kyma-project/kyma: docs/04-operation-guides/troubleshooting/eventing/evnt-03-type-collision.md

## Cause

To conform to Cloud Event specifications, Eventing modifies the event names to filter out prohibited characters. For details, see [event name cleanup](../../../05-technical-reference/evnt-01-event-names.md#event-name-cleanup).
In some cases, it can lead to a naming collision, which can cause subscribers to receive irrelevant events.

## Remedy

Follow these steps to detect if naming collision is the source of the problem:

1. Get the clean types from the status of the Subscription.
 
    ```bash
    kubectl -n {NAMESPACE} get subscriptions.eventing.kyma-project.io {NAME} -o jsonpath='{.status.types}'
    ```

2. Search for any other Subscription using the same `CleanType` as in your Subscription.
    
    ```bash
    kubectl get subscriptions.eventing.kyma-project.io -A | grep {CLEAN_TYPE}
    ```
    

> 3. If you find that the `CleanType` collides with some other Subscription, a solution for this would be to use a different event type. 

Reviewer: ```suggestion
3. If you find that the `CleanType` collides with some other Subscription, a solution for this is to use a different event type. 
```

>     kubectl get subscriptions.eventing.kyma-project.io -A | grep {CLEAN_EVENT_TYPE}

Reviewer 2: Are you sure, this command is still relevant ? 

Also, you could rename the `{CLEAN_EVENT_TYPE}` to `{CLEAN_TYPE}`, since that's the result of the `kubectl -n {NAMESPACE} get subscriptions.eventing.kyma-project.io {NAME} -o jsonpath='{.status.types}'`: 
```
[{"cleanType":"order.paymentsuccess.v1","originalType":"order.payment*success.v1"}]%
```

Author: Updated.
